client/client-multi/client-c/src/platform-impl/kaa_tcp_channel.c

```diff
@@ -305,6 +322,24 @@ kaa_error_t kaa_tcp_channel_sync_handler(void *context, const kaa_service_t serv
         KAA_RETURN_IF_ERR(error_code);
     }
 
+    //If channel pending sync only bootstrap, at sync it should initiate new connection if access point resolved
+    if (((kaa_tcp_channel_t *) context)->pending_request_service_count == 1
```

Comment:
I propose to use the additional "kaa_tcp_channel_t * " - type variable instead of a type cast each time.